cpp/src/arrow/util/compression_zstd.cc

```diff
@@ -215,7 +216,11 @@ Status ZSTDCodec::Decompress(int64_t input_len, const uint8_t* input, int64_t ou
     return ZSTDError(ret, "ZSTD decompression failed: ");
   }
   if (static_cast<int64_t>(ret) != output_len) {
-    return Status::IOError("Corrupt ZSTD compressed data.");
+    std::stringstream ss;
```

Comment:
Can we remove this? The zstd issue is independent of this PR I think :)

Reply:
Oops. It failed in CI before but just been fixed in main branch hours ago. I will remove it.
Another issue is that I don't know how to test it because allocating a huge array will use up CI's memory and it's not a good practice for CI anyway.

Reply:
You can just add a test that is off by default and we can run it on a large machine before we merge this.

Reply:
I think you should be able to rebase and get the zstd fix